### ISSUE TITLE
Utility for adding metals.

### ISSUE THREAD

**Issue by ssepeq** [primaryodors/podock#18]
There is a file called metal.cpp that currently loads a protein and creates an alpha helix in a hard coded location.

The HXXCD/E motif in EXR2 of most ORs is currently the only putative metal binding site thought to form a helix. Its metal should be placed above the ligand binding pocket and the HEAD region moved out of the way if necessary.

This issue is to document ongoing development on the metal binding feature.

(Urgent because a dock results cron is waiting on having better metalloprotein models of the receptors.)

**Comment by ssepeq** [comment]
Look at the 3D structure of TAAR5: http://www.primaryodors.org/or.php?r=TAAR5&tab=Structure

There is a helix above TAAR3 and TAAR5, roughly spanning a line between their respective axes. This helix includes SER180, GLN181, GLU184, and GLU185, which together would strongly attract amines the same way a zinc ion on an ORwould attract most odorants. Below the helix there is a shelf, containing GLN193 which aligns exactly with the two GLU residues of the helix and the space inside the binding pocket, between ASN114 and TRP265, the ideal spot for trimethylamine to do its magic. What's more, TAAR6 thru TAAR9 also have this EE followed by Q motif.

Most ORs, after the HXXCD/E motif, have some variant of a subsequent KLA/SCA/SDT motif. Perhaps the latter also forms a shelf under the helix? Perhaps the helix also spans the space above TMR3 and TMR5? The exact sequence of KLA/SCA/SDT might have a selective effect on incoming potential ligands. For sure, attempts to conformationally fold the EXR2 region have consistently resulted in the metal facing slightly upwards and the region after the HXXCD/E motif to fall below the helix and dip into the binding pocket.

Using the TAARs as an example, see if it's possible to get the HXXCD/E helix into roughly the same position as the TAAR EXR2 helix, and the subsequent region to form a shelf under it. See what part of the KLACADT/KLSCSDT motif aligns to the metal, and where that points to in the binding pocket. If successful, try path docking from the metal, over the shelf, and into the binding pocket, and see what results.

**Comment by ssepeq** [comment]
Can test the new info file using the following commands:

```
wget -O rcpinf.txt http://192.168.0.3/olfaction/recepinfo.php?recep=OR1A1
metal -p /path/to/OR1A1.rotated.pdb -o OR1A1.test.pdb -c 2 -i rcpinf.txt
```

**Comment by ssepeq** [comment]
Can test the new info file using the following commands:

wget -O rcpinf.txt http://www.primaryodors.org/recepinfo.php?recep=OR1A1
metal -p /path/to/OR1A1.rotated.pdb -o OR1A1.test.pdb -c 2 -i rcpinf.txt

**Comment by ssepeq** [comment]
Having difficulty getting this code to place the strand in the desired configuration and avoid atomic collisions.

Decreased the alignfactor delta in Protein::conform_backbone(). Seeing if that helps.

Maybe there is a better way. Perhaps by "guiding" the strand into roughly the right shape, then running a conform() function to resolve atom collisions.

**Comment by ssepeq** [comment]
Most olfactory receptors have an HXXCD/E motif (where X usually = F/Y) in extracellular loop 2 that is thought to bind a metal ion due to the close placement of the H, C, and D/E side chains when coiled into an alpha helix. Source: https://pubmed.ncbi.nlm.nih.gov/12610211/

The motif varies quite a bit between receptors; OR1A1 has NFYCD, OR1D2 has YIFCE, and OR56A4 has NCICS. Despite the variation, all of these are capable of coordinating a metal ion, even the ones using Y since cation-pi binding is possible here.

OR2C1 is one of a few exceptions. It has GFLCE in this position - not suitable for metal binding because glycine lacks a side chain. However, looking slightly outside the motif we find that part of the sequence is actually EGFLCE. If coiled into a pi helix, that should be capable of metal binding. Receptors that have YXXCD/E usually, but not always, have a residue like D or N or H before the Y, and therefore might also have a pi helix for metal coordination rather than relying on the weaker cation-pi bond.

**Comment by ssepeq** [comment]
This will be superseded by #69.
